dart/lib/src/sentry_options.dart

```diff
@@ -423,6 +427,12 @@ typedef BeforeSendCallback = FutureOr<SentryEvent?> Function(
   Hint? hint,
 });
 
+/// This function is called with an SDK specific transaction object and can return a modified transaction
+/// object or nothing to skip reporting the transaction
+typedef BeforeSendTransactionCallback = FutureOr<SentryTransaction?> Function(
+  SentryTransaction transaction,
+);
```

Comment:
```suggestion
typedef BeforeSendTransactionCallback = FutureOr<SentryTransaction?> Function(
  SentryTransaction transaction,
  Hint? hint,
);
```

Reply:
The `captureException` method of the sentry client does not have a hint. https://github.com/getsentry/sentry-dart/blob/b728df439473e146f8721c7555cd54728a93b6cb/dart/lib/src/sentry_client.dart#L292

Is it possible for transaction being captured through he `captureEvent` method? Then it makes sense to also pass on the hint.

Reply:
Good point, we can check the JS/Python implementation, maybe its not needed.
Docs https://github.com/getsentry/sentry-docs/pull/5814

Reply:
Think it does not hurt to at least pass it in the case wehre a potential transaction comes through the captureError method.

Reply:
Just checked other SDKs. Java actually does provide the `Hint` param in the `captureTransaction` method. So we should probably also add this. To the API. WDYT?

Reply:
No need, the `startTransaction` methods don't take a `hint`, so users won't be able to pass it anyway, adding everywhere would be more work, we can add it later if needed.